1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,195 @@
+
+  /**
+   * Get the {@code path} property.
+   * @see #path
+   */
+  public String getPath() { return getString(path); }
+
+  /**
+   * Set the {@code path} property.
+   * @see #path
+   */
+  public void setPath(String v) { setString(path, v, null); }
+
+  //endregion Property "path"
+
+  //region Property "size"
+
+  /**
+   * Slot for the {@code size} property.
+   * @see #getSize
+   * @see #setSize
+   */
+  public static final Property size = newProperty(0, 0, null);
+
+  /**
+   * Get the {@code size} property.
+   * @see #size
+   */
+  public int getSize() { return getInt(size); }
+
+  /**
+   * Set the {@code size} property.
+   * @see #size
+   */
+  public void setSize(int v) { setInt(size, v, null); }
+
+  //endregion Property "size"
+
+  //region Property "createDate"
+
+  /**
+   * Slot for the {@code createDate} property.
+   * @see #getCreateDate
+   * @see #setCreateDate
+   */
+  public static final Property createDate = newProperty(0, BAbsTime.DEFAULT, null);
+
+  /**
+   * Get the {@code createDate} property.
+   * @see #createDate
+   */
+  public BAbsTime getCreateDate() { return (BAbsTime)get(createDate); }
+
+  /**
+   * Set the {@code createDate} property.
+   * @see #createDate
+   */
+  public void setCreateDate(BAbsTime v) { set(createDate, v, null); }
+
+  //endregion Property "createDate"
+
+  //region Property "modifiedDate"
+
+  /**
+   * Slot for the {@code modifiedDate} property.
+   * @see #getModifiedDate
+   * @see #setModifiedDate
+   */
+  public static final Property modifiedDate = newProperty(0, BAbsTime.DEFAULT, null);
+
+  /**
+   * Get the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public BAbsTime getModifiedDate() { return (BAbsTime)get(modifiedDate); }
+
+  /**
+   * Set the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public void setModifiedDate(BAbsTime v) { set(modifiedDate, v, null); }
+
+  //endregion Property "modifiedDate"
+
+  //region Action "touch"
+
+  /**
+   * Slot for the {@code touch} action.
+   * @see #touch()
+   */
+  public static final Action touch = newAction(0, null);
+
+  /**
+   * Invoke the {@code touch} action.
+   * @see #touch
+   */
+  public void touch() { invoke(touch, null, null); }
+
+  //endregion Action "touch"
+
+  //region Action "print"
+
+  /**
+   * Slot for the {@code print} action.
+   * @see #print()
+   */
+  public static final Action print = newAction(0, null);
+
+  /**
+   * Invoke the {@code print} action.
+   * @see #print
+   */
+  public void print() { invoke(print, null, null); }
+
+  //endregion Action "print"
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BFile.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+
+    @SuppressWarnings("unused")
+	public void doTouch() {
+		setModifiedDate(BAbsTime.make());
```

Comment:
I'd prefer to use `BAbsTime.now()` instead of relying on the behavior of `make` when you don't pass in any parameters, but this is fine.